net-utils/src/lib.rs

```diff
@@ -384,29 +384,67 @@ pub fn is_host_port(string: String) -> Result<(), String> {
     parse_host_port(&string).map(|_| ())
 }
 
+#[derive(Clone, Debug)]
+pub struct SocketConfig {
+    pub reuseaddr: bool,
+    pub reuseport: bool,
+}
+
+impl Default for SocketConfig {
+    #[allow(clippy::derivable_impls)]
+    fn default() -> Self {
+        Self {
+            reuseaddr: false,
+            reuseport: false,
+        }
+    }
+}
+
 #[cfg(any(windows, target_os = "ios"))]
 fn udp_socket(_reuseaddr: bool) -> io::Result<Socket> {
     let sock = Socket::new(Domain::IPV4, Type::DGRAM, None)?;
     Ok(sock)
 }
 
+#[cfg(any(windows, target_os = "ios"))]
+fn udp_socket_with_config(_config: SocketConfig) -> io::Result<Socket> {
+    let sock = Socket::new(Domain::IPV4, Type::DGRAM, None)?;
+    Ok(sock)
+}
+
 #[cfg(not(any(windows, target_os = "ios")))]
 fn udp_socket(reuseaddr: bool) -> io::Result<Socket> {
+    let config = SocketConfig {
+        reuseaddr,
+        reuseport: false,
+    };
+    udp_socket_with_config(config)
+}
+
+#[cfg(not(any(windows, target_os = "ios")))]
+fn udp_socket_with_config(config: SocketConfig) -> io::Result<Socket> {
     use {
         nix::sys::socket::{
             setsockopt,
             sockopt::{ReuseAddr, ReusePort},
         },
         std::os::unix::io::AsRawFd,
     };
+    let SocketConfig {
```

Comment:
We were creating multiple UDP Socket before with reuseaddr flag and seems to work. What makes it necessary for Quic's socket we need to explicitly do reuse port?

Reply:
What I meant is in the original code it was already does SO_REUSEPORT: 

```
    if reuseaddr {
         // best effort, i.e. ignore errors here, we'll get the failure in caller
         setsockopt(sock_fd, ReusePort, &true).ok();
         setsockopt(sock_fd, ReuseAddr, &true).ok();
     }
```